### PR TITLE
HPCC-27745 allow each data plane has its own storage account

### PR DESCRIPTION
**Pull request by xwang2713**
(secretName) and namespace

Sample usage:
common:
  mountPrefix: "/var/lib/HPCCSystems"
  secretName: "azure-secret"
  secretNamespace: "default"
  provisioner: "file.csi.azure.com" 

planes:
- name: dali
  subPath: dalistorage
  size: 1Gi
  category: dali
  shareName: dalishare
  secretName: daliscsecret
.......

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Test with "helm template ..."

### CODE REVIEW

**Review by dcamper — APPROVED** [review]

**Comment by dcamper** [comment]
@xwang2713 Do you need to increment the versions in Chart.yaml?

File: helm/examples/azure/hpcc-azurefile/values.schema.json [hunk 1]
```diff
@@ -71,6 +71,14 @@
         "shareName": {
           "description": "Azure storage share name",
           "type": "string"
+        },
+        "secretName": {
+          "description": "the secretName, the storage account, for current plane",
```
Comment (jakesmith) [inline helm/examples/azure/hpcc-azurefile/values.schema.json:76]: I think better reworded to be like the common description this overrides, e.g.:

"Kubernetes secret name to use for the Azure storage account for this plane"

File: helm/examples/azure/hpcc-azurefile/values.schema.json [hunk 2]
```diff
@@ -71,6 +71,14 @@
...
+          "type": "string"
+        },
+        "secretNamespace": {
+          "description": "the namespace of the secret, storage account",
```
Comment (jakesmith) [inline helm/examples/azure/hpcc-azurefile/values.schema.json:80]: similarly:
"Kubernetes namespace for the Kubernetes secret named by secretName"

**Review by jakesmith — COMMENTED** [review]
@xwang2713 - minor comments.

I agree with @dcamper - version should be bumped.

Also, minor - but I think as there are example commented out secretName and secretNamespace under common in values.yaml, it would be good if was one plane that also had a commented out example.

**Review by ghalliday — APPROVED** [review]
no extra comments

File: helm/examples/azure/hpcc-azurefile/values.schema.json (hunk 1 above)
Comment (xwang2713) [reply helm/examples/azure/hpcc-azurefile/values.schema.json:76]: Fixed

File: helm/examples/azure/hpcc-azurefile/values.schema.json (hunk 2 above)
Comment (xwang2713) [reply helm/examples/azure/hpcc-azurefile/values.schema.json:80]: Fixed

**Comment by xwang2713** [comment]
Increased the chart version to "0.2.0"

**Review by jakesmith — COMMENTED** [review]
@xwang2713 approved, but you may have missed my last comment:

"Also, minor - but I think as there are example commented out secretName and secretNamespace under common in values.yaml, it would be good if was one plane that also had a commented out example."

**Review by dcamper — APPROVED** [review]

**Comment by xwang2713** [comment]
@jakesmith add an example of secretName and namespaceName for Dali plane and rebased.

**Review by jakesmith — APPROVED** [review]
@xwang2713 - looks good

**Pull request merged**